### ISSUE TITLE
design suggestion: formatters in analysers

### ISSUE THREAD

**Issue by tykayn** [osm-fr/osmose-backend#2054]
i saw many conversions in analyses that could be gathered instead of repeating the formatting functions.

for example, phone numbers could have their format being converted to a spaced and international prefixed format with one function of the Analyser_Merge class.
This could be done also by a separated class that would do the formatting of the input_string to a desired OSM field.

for example i would import the Formatting class in my analyser and do the field mapping like so:

```python
# example 
class Analyser_Merge_Parking_FR_capp(Analyser_Merge):
    def __init__(self, config, logger = None):
       # ...
        self.init(
       # ...
            mapping = Mapping(
# ...
                    mapping1 = {
                        "name": "NOM",
                        "capacity": Formatter( res["Places"]  , 'number_positive') ,
# instead of : "capacity": lambda res: res["Places"] if res["Places"] != "0" else None,
# or from the Analyser_Merge
                        "capacity": self.formatter( res["Places"]  , 'number_positive') ,
```

this could be very useful to have an homogenous validation constraint on all the Analaysers,
could be great for opening hours and many common things.

**Comment by frodrigo** [comment]
There is some utilities in https://github.com/osm-fr/osmose-backend/tree/dev/modules

**Comment by Famlam** [comment]
For the specific case on the phone numbers, it sounds nice, but to do it properly, it would require us to have a list to:
1. detect (and exclude) short numbers (numbers that cannot be internationalized directly, such as 112, or the `12***`-series in NL)
2. add the country code if omitted, thereby taking into account whether the 0 has to be omitted (most countries) or kept (e.g. Italy, see https://en.wikipedia.org/wiki/Telephone_numbers_in_Italy#:~:text=The%20dialing%20procedures%20for%20callers,practice%20in%20many%20other%20countries. )
3. Then have a list of the exact structure of the (various) phone number formats per region to decide where the spaces should be. Taking NL as an example, we have regions where the local code is 3 digits or 4 digits, so you'd need a city-level database to determine whether you should format a city as +31 NN NNNNNNN OR +31 NNN NNNNNN (https://nl.wikipedia.org/wiki/Lijst_van_Nederlandse_netnummers#/media/Bestand:2010-Telefonie-Prov-650px.png ). In addition, asides from the regional phone numbers, NL has phone numbers that work with +31 varying from `+31 [+7 digits]` to `+31 [+11 digits]` length. So where to put the spaces will be very difficult too. (Assuming the NL example holds worldwide)

It sounds easier than it probably is

**Comment by frodrigo** [comment]
Note about phone number, there is a formatting lib provided by Google.
The phone plugin does not use it because we does not want to report about format changing only.

**Comment by Famlam** [comment]
> The phone plugin does not use it because we does not want to report about format changing only.

Agree, but for the merge analysers (or places where we suggest fixes) we can just as well format it in the proposed addition :)
